network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -312,106 +134,149 @@ void prepareStatements() {
+            connection.setAutoCommit(false);
+            try {
+                createNetworks(connection, resources);
+            } finally {
+                connection.setAutoCommit(true);
             }
-            batch = batch.addAll(boundStatements);
-            session.execute(batch);
+        } catch (SQLException e) {
+            throw new UncheckedSqlException(e);
         }
     }
 
-    public void updateNetworks(List<Resource<NetworkAttributes>> resources) {
-        Map<String, Mapping> networkMappings = mappings.getNetworkMappings().getColumnMapping();
-        Set<String> keysNetworks = networkMappings.keySet();
-        PreparedStatement psUpdateNetwork = updatePreparedStatements.get(NETWORK);
-
-        for (List<Resource<NetworkAttributes>> subresources : Lists.partition(resources, BATCH_SIZE)) {
-            BatchStatement batch = BatchStatement.newInstance(BatchType.UNLOGGED);
-            List<BoundStatement> boundStatements = new ArrayList<>();
-            for (Resource<NetworkAttributes> resource : subresources) {
-                NetworkAttributes networkAttributes = resource.getAttributes();
-                List<Object> values = new ArrayList<>();
-                values.add(resource.getId());
-                keysNetworks.forEach(key -> {
-                    if (!key.equals(UUID_STR) && !key.equals(VARIANT_ID)) {
-                        values.add(networkMappings.get(key).get(networkAttributes));
+    private void createNetworks(Connection connection, List<Resource<NetworkAttributes>> resources) throws SQLException {
+        var tableMapping = mappings.getNetworkMappings();
+        try (var preparedStmt = connection.prepareStatement(QueryCatalog.buildInsertNetworkQuery(tableMapping.getTable(), tableMapping.getColumnMapping().keySet()))) {
+            List<Object> values = new ArrayList<>(2 + tableMapping.getColumnMapping().size());
+            for (List<Resource<NetworkAttributes>> subResources : Lists.partition(resources, BATCH_SIZE)) {
+                for (Resource<NetworkAttributes> resource : subResources) {
+                    NetworkAttributes attributes = resource.getAttributes();
+                    values.clear();
+                    values.add(resource.getVariantNum());
+                    values.add(resource.getId());
+                    for (var mapping : tableMapping.getColumnMapping().values()) {
+                        values.add(mapping.get(attributes));
                     }
-                });
-                values.add(networkAttributes.getUuid());
-                values.add(resource.getVariantNum());
+                    bindValues(preparedStmt, values);
+                    preparedStmt.addBatch();
+                }
+                preparedStmt.executeBatch();
+            }
+        }
+    }
 
-                boundStatements.add(psUpdateNetwork.bind(values.toArray(new Object[0])));
+    public void updateNetworks(List<Resource<NetworkAttributes>> resources) {
```

Comment:
use the same split structure as createNetworks ?

Reply:
Even for createNetworks, it is useless to split as there will be only a few numbers of resources.

Reply:
looking at the code resources are split into batches?

Reply:
No i meant Split the code like you did in 2 méthodes.